src/overlays/actors/ovl_Bg_Mizu_Bwall/z_bg_mizu_bwall.c

```diff
@@ -21,23 +28,435 @@ const ActorInit Bg_Mizu_Bwall_InitVars = {
+    Vec3f debrisOffsets[15];
+
+    for (i = 0; i < ARRAY_COUNT(debrisOffsets); i++) {
+        switch ((u16)this->dyna.actor.params & 0xF) {
+            case MIZUBWALL_FLOOR:
+                debrisOffsets[i].x = (Rand_ZeroOne() * 80.0f) - 40.0f;
+                debrisOffsets[i].y = Rand_ZeroOne() * 0;
+                debrisOffsets[i].z = (Rand_ZeroOne() * 80.0f) - 40.0f;
+                break;
+            case MIZUBWALL_RUTO_ROOM:
+                debrisOffsets[i].x = Rand_ZeroOne() * 0;
+                debrisOffsets[i].y = Rand_ZeroOne() * 100.0f;
+                debrisOffsets[i].z = (Rand_ZeroOne() * 80.0f) - 40.0f;
+                break;
+            case MIZUBWALL_UNUSED:
+            case MIZUBWALL_STINGER_ROOM_1:
+            default:
+                debrisOffsets[i].x = (Rand_ZeroOne() * 120) - 60.0f;
+                debrisOffsets[i].y = Rand_ZeroOne() * 120;
+                debrisOffsets[i].z = Rand_ZeroOne() * 0;
+                break;
+        }
+    }
+
+    for (i = 0; i < ARRAY_COUNT(debrisOffsets); i++) {
+        tempx = debrisOffsets[i].x;
+        tempz = debrisOffsets[i].z;
+
+        debrisPos.x = thisPos->x + tempz * sin + tempx * cos;
+        debrisPos.y = thisPos->y + debrisOffsets[i].y;
+        debrisPos.z = thisPos->z + tempz * cos - tempx * sin;
+
+        rand1 = (s16)(Rand_ZeroOne() * 120.0f) + 20;
+        rand2 = (s16)(Rand_ZeroOne() * 240.0f) + 20;
+        func_80033480(globalCtx, &debrisPos, 50.0f, 2, rand1, rand2, 0);
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_A_OBJ, debrisPos.x, debrisPos.y, debrisPos.z, 0, 0, 0,
+                    0xB);
+    }
+}
+
+void BgMizuBwall_Idle(BgMizuBwall* this, GlobalContext* globalCtx) {
+    BgMizuBwall_SetAlpha(this, globalCtx);
+    if (this->collider.base.acFlags & 2) {
+        this->collider.base.acFlags &= ~2;
+        Flags_SetSwitch(globalCtx, ((u16)this->dyna.actor.params >> 8) & 0x3F);
+        this->breakTimer = 1;
+        func_8003EBF8(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+        this->dList = NULL;
+        BgMizuBwall_SpawnDebris(this, globalCtx);
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_WALL_BROKEN);
+        Audio_PlaySoundGeneral(NA_SE_SY_CORRECT_CHIME, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        this->actionFunc = BgMizuBwall_Break;
+    } else if (this->dyna.actor.xzDistFromLink < 600.0f) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mizu_Bwall/BgMizuBwall_Destroy.s")
+void BgMizuBwall_Break(BgMizuBwall* this, GlobalContext* globalCtx) {
+    if (this->breakTimer > 0) {
+        this->breakTimer--;
+    } else {
+        this->actionFunc = BgMizuBwall_Destroyed;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mizu_Bwall/func_8089CE04.s")
+void BgMizuBwall_Destroyed(BgMizuBwall* this, GlobalContext* globalCtx) {
+}
```

Comment:
id say DoNothing for this name personally. just cause its literally doing nothing, and it can be confused with the Destroy function at a quick glance lol